atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -62,16 +63,21 @@ def run(self):
 
             base_image_with_registry.registry = self.parent_registry
 
-        pulled_base = self.tasker.pull_image(base_image_with_registry,
-                                             insecure=self.parent_registry_insecure)
-        if (base_image_with_registry.namespace != 'library' and
-                not self.tasker.image_exists(base_image_with_registry.to_str())):
-            self.log.info("'%s' not found", base_image_with_registry.to_str())
-            base_image_with_registry.namespace = 'library'
-            self.log.info("trying '%s'", base_image_with_registry.to_str())
-            pulled_base = self.tasker.pull_image(base_image_with_registry,
-                                                 insecure=self.parent_registry_insecure)
+        try:
+            self.tasker.pull_image(base_image_with_registry,
+                                   insecure=self.parent_registry_insecure)
 
+        except RetryGeneratorException:
+            if base_image_with_registry.namespace != 'library':
```

Comment:
Why is `not self.tasker.image_exists(base_image_with_registry.to_str()))` omitted here?

Reply:
pull will now raise exceptions

Reply:
Could you elaborate?

Reply:
well we don't have to check image existence anymore because if 'pull'  is successful it won't be raising any error, but if there is problem it will raise, unlike before when it was just logging errors but didn't stop

Reply:
Ah, this part is removed as `image_exists` is assumed to be True if `pull_image` has worked